projects/components/src/table/cells/table-cell-parser-base.ts

```diff
@@ -0,0 +1,17 @@
+import { TableRow } from '../table-api';
+import { CoreTableCellParserType } from './types/core-table-cell-parser-type';
+
+export abstract class TableCellParserBase<TCellData, TValue, TFilterValue> {
+  public static readonly type: CoreTableCellParserType;
+
+  public abstract parseValue(cellData: TCellData, rowData: TableRow): TValue;
+  public abstract parseFilterValue(cellData: TCellData): TFilterValue;
```

Comment:
shouldn't this default to just parseValue? Seems like that's its implementation almost everywhere.

Reply:
Currently there are 9 cell renderers that don't require override and 6 that do. Since the majority of the renderers that _don't_ require the override and can just use `parseValue` directly are the primitive types I think I'd like to keep it abstract. Most, if not all, renderers we add from now on will be composite and require the `parseFilterValue` method.

I think having users see they need to implement it will prevent them from forgetting it and causing a filtering issue.